### DISCUSSION TITLE
Question about License?

### DISCUSSION THREAD

**Discussion by AlbertShown** [cesanta/mongoose#2135] [Ask questions, get answers] [answered]
https://www.gnu.org/licenses/rms-why-gplv3.en.html

> When we say that GPLv2 and GPLv3 are incompatible, it means there is no legal way to combine code under GPLv2 with code under GPLv3 in a single program. This is because both GPLv2 and GPLv3 are copyleft licenses: each of them says, “**_If you include code under this license in a larger program, the larger program must be under this license too._**” There is no way to make them compatible.

Is that mean if our program/library use Mongoose source code, then we have no choice other than to use the same Mongoose License, which is `GPLv2`?

**Comment by guest271314** [comment]
I am also interested in the license relevant to using mongoose as a shared library in QuickJS https://github.com/bellard/quickjs/blob/master/LICENSE.

**Comment by novlean** [comment, marked as answer]
When integrating Mongoose Library, you agree to the license terms under the GPLv2 license. 
Incompatibilities, through taking different components of different sources under different license restrictions, can result in the software pieces you choose not to be permitted to be used in a single product. 
If you are unsure on the different open source licenses compatibility, please seek independent and qualified legal advice on this matter.

For commercial applications, a commercial license shall be considered. 
Commercial license removes all GPLv2 restrictions and allows you to redistribute your closed source products with the Mongoose Library embedded.

**Comment by AlbertShown** [comment]
I understand, thank you. I will use then GPLv2.
